userguide/tutorials/invoice_examples.adoc

```diff
@@ -0,0 +1,169 @@
+
+. As soon as a subscription is created, it is in `TRIAL` phase. An invoice is generated with one invoice item of type `FIXED` and amount `0`. 
+
+So, `CHARGED AMOUNT` and `BALANCE` are calculated as follows:
+
+[source,java]
+----
+CHARGED AMOUNT = FIXED + RECURRING + EXTERNAL_CHARGE + USAGE + TAX + ITEM_ADJ + REPAIR_ADJ + CREDIT_ADJ
+               = 0 + 0 + 0 + 0 + 0 + 0 + 0
+               = 0
+               
+INVOICE BALANCE = CHARGED AMOUNT - AMOUNT PAID - AMOUNT REFUNDED - SUM(CBA_ADJ)
+                = 0 - 0 - 0 - 0
+                = 0                
+----
+
+[start=2]
+. Once the `TRIAL` phase ends, the subscription moves to the `EVERGREEN` phase. An invoice is generated with one invoice item of type `RECURRING` and amount `24.95`.
+
+So, `CHARGED AMOUNT` and `BALANCE` are calculated as follows:
+
+[source,java]
+----
+CHARGED AMOUNT = FIXED + RECURRING + EXTERNAL_CHARGE + USAGE + TAX + ITEM_ADJ + REPAIR_ADJ + CREDIT_ADJ
+               = 0 + 24.95 + 0 + 0 + 0 + 0 + 0 + 0
+               = 24.95
+               
+INVOICE BALANCE = CHARGED AMOUNT - AMOUNT PAID - AMOUNT REFUNDED - SUM(CBA_ADJ)  
+                = 24.95 - 24.95 - 0 - 0   // assuming the account has a valid default payment method, AMOUNT PAID=24.95
+                = 0
+----
+             
+== Adding an invoice credit
+
+Whenever a https://killbill.github.io/slate/#credit-create-credits[credit is added], the system generates an invoice credit. An invoice credit is nothing but an invoice with a `CREDIT_ADJ` and a `CBA_ADJ` item. When such a credit is added, it is adjusted against the amount/balance in other unpaid invoices (if any). 
+
+Let us understand what happens in this scenario:
+
+. When a credit is added (let's say for $20), an invoice is generated with two invoice items: `Type=CREDIT_ADJ, Amount=-20`, and `Type=CBA_ADJ, Amount=20`. The `CREDIT_ADJ` invoice item corresponds to the credit and is negative while the `CBA_ADJ` item is positive and is automatically inserted by the system to bring the invoice balance to `0`.
+
+So, `CHARGED AMOUNT` and `BALANCE` are calculated as follows:
+
+[source,java]
+----
+CHARGED AMOUNT = FIXED + RECURRING + EXTERNAL_CHARGE + USAGE + TAX + ITEM_ADJ + REPAIR_ADJ // CREDIT_ADJ not taken into account for invoice credit
+               = 0 + 0 + 0 + 0 + 0 + 0 + 0 
+               = 0
+               
+INVOICE BALANCE = CHARGED AMOUNT - AMOUNT PAID - AMOUNT REFUNDED - SUM(CBA_ADJ)  + SUM(CREDIT_ADJ) //CREDIT_ADJ taken into account for invoice credit
+                = 0 - 0 - 0 + 20 + (-20)
```

Comment:
Ah ah, so the formula is indeed incorrect: `- SUM(CBA_ADJ) ` -> `- (20)` -> you would end up with `-40`

Reply:
Yes, fixed